tests/functional_tests/conftest.py

```diff
@@ -90,21 +91,48 @@ def kube_client(request, in_cluster=False):
     Return a kubernetes client. By default, use kube-config. If running in a pod, use k8s service account.
     """
 
+    k8s_client = get_kube_client(in_cluster)
+    yield k8s_client
+
+
+def get_kube_client(in_cluster=False):
     if in_cluster:
         print("Using in cluster kubernetes configuration")
         config.load_incluster_config()
     else:
         print("Using kubectl kubernetes configuration")
         config.load_kube_config()
-    yield client.CoreV1Api()
 
+    return client.CoreV1Api()
 
-def get_pod_by_label_selector(kube_client, label_selector, namespace=namespace) -> str:
+
+def get_pod_by_label_selector(
+    kube_client, label_selector, pod_namespace=namespace
+) -> str:
     """Return the name of a pod found by label selector."""
     pods = kube_client.list_namespaced_pod(
-        namespace, label_selector=label_selector
+        pod_namespace, label_selector=label_selector
     ).items
     assert (
         len(pods) > 0
     ), f"Expected to find at least one pod with labels '{label_selector}'"
     return pods[0].metadata.name
+
+
+def get_pod_running_containers(pod_namespace=namespace):
+    """Return the containers from pods found."""
+    pods = get_kube_client().list_namespaced_pod(pod_namespace).items
+
+    containers = {}
+    for pod in pods:
+        pod_name = pod.metadata.name
+        for container_status in pod.status.container_statuses:
+            if container_status.ready:
+                container_name = container_status.name
+                containers[container_name] = {
+                    "container": vars(container_status).copy(),
+                    "namespace": pod_namespace,
+                    "pod_name": pod_name,
+                }
```

Comment:
@danielhoherd Now we return dict instead of the list. That will help to get ids in pytest parameters.